src/components/ZetkinSpeedDial/actions/createTask.tsx

```diff
@@ -0,0 +1,40 @@
+/* eslint-disable react/display-name */
+import { ACTIONS } from '../constants';
+import { CheckBox } from '@material-ui/icons';
+import { useMutation } from 'react-query';
+import { useRouter } from 'next/router';
+
+import CreateTaskForm from '../../organize/tasks/forms/CreateTaskForm';
+import postTask from '../../../fetching/tasks/postTask';
+import { ZetkinTaskReqBody } from '../../../types/zetkin';
+import { ActionConfig, DialogContentBaseProps } from './types';
+
+
+const DialogContent: React.FunctionComponent<DialogContentBaseProps> = ({ closeDialog }) => {
+    const router = useRouter();
+    const { campId, orgId } = router.query as {campId: string; orgId: string};
+
+    const eventMutation = useMutation(postTask(orgId));
+
+    const handleFormSubmit = async (data: ZetkinTaskReqBody) => {
+        closeDialog();
+        const newTask = await eventMutation.mutateAsync(data);
+        // Redirect to task page
+        router.push(`/organize/${orgId}/campiagns/${campId}/tasks/${newTask.id}`);
```

Comment:
Nice and simple solution!

Reply:
Can be used for the campaign redirect on submit issue as well 🆒